types/src/state_store/state_value.rs

```diff
@@ -178,14 +179,34 @@ impl PersistedStateValue {
     }
 }
 
-#[derive(BCSCryptoHash, Clone, CryptoHasher, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, BCSCryptoHash, CryptoHasher)]
 pub struct StateValue {
     data: Bytes,
     metadata: StateValueMetadata,
+    maybe_rapid_hash: Option<(u64, usize)>,
 }
 
+impl PartialEq for StateValue {
+    fn eq(&self, other: &Self) -> bool {
+        // Fast path: if both have rapid hashes and they differ, values can't be equal
+        if let (Some(hash1), Some(hash2)) = (&self.maybe_rapid_hash, &other.maybe_rapid_hash) {
+            if hash1 != hash2 {
+                return false;
+            }
+        }
+
+        // Full comparison: data and metadata
+        self.data == other.data && self.metadata == other.metadata
```

Comment:
do you know if comparison of slices here first checks for length, before looking at bytes?